controllers/articlesCommentController.js

```diff
@@ -85,8 +95,101 @@ async function findCommentThreadController(req, res) {
   }
 }
 
+
+/**
+ *
+ * @description update Article Comment Controller
+ * @param {*} req
+ * @param {*} res
+ * @returns {*} *
+ */
+async function updateCommentController(req, res) {
+  try {
+    const userId = req.decodedToken.payLoad;
+    const { commentId, comment } = req.body;
+    const { articleId } = req.params;
+    const commenter = await Comment.findComment(articleId, commentId);
+    await Comment.updateArticleComment({ updated: true }, commentId);
+    const createComment = await Comment.createComment({
+      comment,
+      userId,
+      articleId,
+      mainCommentId: commenter.mainCommentId,
+      createdAt: commenter.createdAt
+    });
+    if (!commenter.mainCommentId) {
+      await Comment.updateAllCommentThreads({
+        mainCommentId: createComment.id
+      }, commentId);
+    }
+    return successResponse(res, COMMENT_UPDATED, RESOURCE_CREATED_CODE, createComment);
+  } catch (error) {
+    return serverFailure(res, SERVER_ERROR_MESSAGE);
+  }
+}
+
+/**
+ * @description Gets all comment history Controller
+ * @param {*} req
+ * @param {*} res
+ * @returns {*} *
+ */
+async function getAllHistory(req, res) {
+  try {
+    const history = await commentHistory.getAllHistory();
+    const commentCount = history.count,
+      comments = history.rows;
+    const result = { commentCount, comments };
+    successResponse(res, HISTORY_RETRIEVED, OK_CODE, result);
+  } catch (error) {
+    return serverFailure(res, SERVER_ERROR_MESSAGE);
+  }
+}
+
+/**
+ * @description Gets comment history under a specific article Controller
+ * @param {*} req
+ * @param {*} res
+ * @returns {*} *
+ */
+async function getArticlesHistory(req, res) {
```

Comment:
Please explain why we have `getArticlesHistory`?

Reply:
Get all comments updated or deleted from an article